Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp

```diff
@@ -515,6 +514,24 @@ bool RemoteRenderingBackendProxy::isCached(const ImageBuffer& imageBuffer) const
     return false;
 }
 
+void RemoteRenderingBackendProxy::imageBufferSetDidPrepareForDisplay(WebKit::RemoteImageBufferSetIdentifier bufferSet, WebKit::ImageBufferSetPrepareBufferForDisplayOutputData outputData)
+{
+    m_background->putImageBufferSetPrepareForDisplayOutputData(bufferSet, WTFMove(outputData));
```

Comment:
That's fair. The complication is that we need to keep those `RemoteImageBufferSetProxy` objects alive while we wait, since the main-thread will be free to run other code, which might remove layers.